modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedValueProviderCacheIdGenerator.java

```diff
@@ -101,24 +103,32 @@ public DslElementBasedValueProviderCacheIdGenerator(ComponentLocator<DslElementM
                                                                  Map<String, ParameterModelInformation> parameterModelsInformation) {
     List<ValueProviderCacheId> parts = new LinkedList<>();
 
-    parts.add(resolveValueProviderId(valueProviderModel));
     parts.addAll(resolveActingParameterIds(valueProviderModel, parameterModelsInformation));
+    parts.add(resolveValueProviderId(valueProviderModel));
+    parts.add(aValueProviderCacheId(fromElementWithName(VALUE_PROVIDER).withHashValueFrom(VALUE_PROVIDER)));
 
     String id = resolveDslTag(containerComponent).orElse(getSourceElementName(containerComponent));
-    return of(aValueProviderCacheId(fromElementWithName(id).withHashValueFrom(id).containing(parts)));
+    return of(aValueProviderCacheId(fromElementWithName(id)
+        .withHashValueFrom(resolveDslTagNamespace(containerComponent).orElse(id)).containing(parts)));
   }
 
   private Optional<ValueProviderCacheId> resolveForComponentModel(DslElementModel<?> containerComponent,
                                                                   ValueProviderModel valueProviderModel,
                                                                   Map<String, ParameterModelInformation> parameterModelsInformation) {
     List<ValueProviderCacheId> parts = new LinkedList<>();
 
-    parts.add(resolveValueProviderId(valueProviderModel));
     parts.addAll(resolveActingParameterIds(valueProviderModel, parameterModelsInformation));
     parts.addAll(resolveIdForInjectedElements(containerComponent, valueProviderModel));
+    parts.add(resolveValueProviderId(valueProviderModel));
+    parts.add(aValueProviderCacheId(fromElementWithName(VALUE_PROVIDER).withHashValueFrom(VALUE_PROVIDER)));
 
     String id = resolveDslTag(containerComponent).orElse(getSourceElementName(containerComponent));
-    return of(aValueProviderCacheId(fromElementWithName(id).withHashValueFrom(id).containing(parts)));
+    return of(aValueProviderCacheId(fromElementWithName(id)
```

Comment:
only the namespace and if not present the id? You are assuming that if the namespace is not present, then the id should be the identifier right?

Reply:
Yes, there is no other way to associate it to an extension.